main.go

```diff
@@ -91,55 +91,35 @@ func wrapMain() error {
 
 	fmt.Printf("Drone GAE Plugin built from %s\n", rev)
 
-	// https://godoc.org/github.com/drone/drone-plugin-go/plugin
-	workspace := plugin.Workspace{}
-	repo := plugin.Repo{}
-	build := plugin.Build{}
-	system := plugin.System{}
 	vargs := GAE{}
+	workspace := ""
 
-	plugin.Param("workspace", &workspace)
-	plugin.Param("repo", &repo)
-	plugin.Param("build", &build)
-	plugin.Param("system", &system)
-	plugin.Param("vargs", &vargs)
-	plugin.MustParse()
-
-	// Check required params
-	if vargs.Token == "" {
-		return fmt.Errorf("missing required param: token")
-	}
-
-	if vargs.Project == "" {
-		vargs.Project = getProjectFromToken(vargs.Token)
-	}
-
-	if vargs.Project == "" {
-		return fmt.Errorf("missing required param: project")
+	// Check what drone version we're running on
+	if os.Getenv("DRONE_WORKSPACE") == "" { // 0.4
+		err := configFromStdin(&vargs, &workspace)
+		if err != nil {
+			return err
+		}
+	} else { // 0.5+
+		err := configFromEnv(&vargs, &workspace)
```

Comment:
this paradigm of passing pointers to things that the function writes into (vs return values) is new to me, so let me know if this isn't right.